modules/seqexec/engine/src/main/scala/seqexec/engine/Sequence.scala

```diff
@@ -192,7 +192,13 @@ object Sequence {
 
     def start(i: Int): State
 
-    def update(sequence: Sequence): State
+    /**
+      * Updates the steps executions.
+      * It preserves the number of steps.
+      * @param stepDefs New executions.
+      * @return Updated state
+      */
+    def update(stepDefs: List[List[Actions]]): State
```

Comment:
I've always found these a bit confusing with the nested `List` I wonder if there is a better way to express this

Reply:
I could create type aliases

Reply:
What's wrong with a list of lists? It's very clear what it means.

Reply:
the val name `stepDeps` helps somewhat but perhaps another type could moke it clearer.